### PR TITLE
Feat 1266 credit class page project cards

### PR DESCRIPTION
**Pull request by victordelval**
## Description

Closes: regen-network/regen-registry#1266

(*) This branch comes out of another development branch, see PR: https://github.com/regen-network/regen-web/pull/1232

Covers the on-chain case (both credit classes and projects)
* Extract component for Projects section
* Use new hooks and card components with Purchase enabled
* Add credit class filter to the hook that prepares projects with sell orders

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1. https://deploy-preview-1339--regen-registry.netlify.app/credit-classes/C01
2. https://deploy-preview-1339--regen-registry.netlify.app/credit-classes/C45

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/pages/CreditClassDetails/CreditClassDetails.Projects.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { ProjectWithOrderData } from 'pages/Projects/Projects.types';
+import { ProjectCardsSection } from 'components/organisms/ProjectCardsSection/ProjectCardsSection';
+import { useProjectsWithOrders } from 'hooks/projects/useProjectsWithOrders';
+
+interface Props {
+  classId: string;
+}
+
+function ProjectsSection({ classId }: Props): JSX.Element {
+  const { projectsWithOrderData } = useProjectsWithOrders({
```
Comment (flagrede) [inline web-registry/src/pages/CreditClassDetails/CreditClassDetails.Projects.tsx:14]: This hook uses the `projects` method which query all the projects. In this case, we should use [ProjectsByClass](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.Query.ProjectsByClass) instead.
We can either modify the existing hook here or duplicate it if it's easier.

**Review by flagrede — CHANGES_REQUESTED** [review]
Nice !
Just one request change the change the query.

File: web-registry/src/pages/CreditClassDetails/CreditClassDetails.Projects.tsx (hunk 1 above)
Comment (victordelval) [reply web-registry/src/pages/CreditClassDetails/CreditClassDetails.Projects.tsx:14]: @flagrede Good point, thanks! I just do initial check to launch the specific initial query. I remove post filtering.

**Review by flagrede — APPROVED** [review]
Looks good !

**Review by wgwz — APPROVED** [review]
LGTM!

**Comment by victordelval** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**